### ISSUE TITLE
Release un-minified version of js client

### ISSUE THREAD

**Issue by futuro** [google/google-api-javascript-client#211]
Can we get an un-minified, readable version of the js client? The documentation on developers.google.com is chaotic/semi-inscrutable and I have, several times, tried to read through the client.js file to understand what's going on. Having a human-readable version would be helpful.

**Comment by yk1711** [comment]
+1

**Comment by trustedjohn** [comment]
+1!

**Comment by ianschmitz** [comment]
+1

**Comment by sergiocallegari** [comment]
Possibly api.js too? This seems to be completely undocumented.

**Comment by bsittler** [comment]
I'm certainly excited to see the interest here! What goal (or goals) are you hoping to reach more easily by way of un-minified, readable source for the js client and api.js? Are there any specific shortcomings in the documentation you would like to see addressed?

**Comment by futuro** [comment]
Unfortunately back when I wrote this up I didn't give concrete examples of what I hoped to achieve, so the exact details have escaped me, but here's what I do remember.

Navigating the documentation was an IX nightmare, from finding the documentation for dealing with what part of the Google suite I wanted to work with, to then understanding how the various bits of those libraries fit together. For example, to get a users profile info I came across documentation that talked about `gapi.profile` but that didn't work, then I found `gapi.plus` which also failed, and ultimately I discovered that it's under `gapi.client.oauth2.userinfo.get`. While this makes sense in a way, the path to finding it felt like a fever dream as I jumped from doc page to doc page.

Ultimately the issues I've run into boil down to a lack of sign-posting and an information structure that I can't really parse and doesn't ultimately seem to have any apparent hierarchy. I don't mean that it doesn't, per se, or that people haven't put time and effort into it, but that I couldn't ever figure out what the hierarchy was supposed to be or how to place each doc section in context with the rest of the docs. I ultimately ended up memorizing what different docs for dealing with the calendar api looked like and bookmarking them, because they both had a mix of accurate and inaccurate data.

So part of what I hope for from having legible source is to be able to work around the IX woes I had with the docs, and then also for easier discoverability. For example, it's not really clear what values `client.load` accepts, and there's nowhere online that I could find which laid it out. Having the source might help with that.

I'd also mention that the fact that some products, such as the calendar, live under the `gapi.client` namespace while others, like youtube, seem to live just under `gapi` adds to the confusion, as I can't discern why calendar is considered "client" code but youtube isn't.

That's a bit long, but I hope it lays out what I'd like to get from an un-obfuscated source release. Please let me know if there's anything I can help with.

Thank you

P.S. As a side note for anyone having trouble with the auth libraries coming from a google search, know that you need to load platform.js with `https://apis.google.com/js/client:platform.js`, then `gapi.load` the "auth2" lib. Top level user info is in the `gapi.client.oauth2` lib (use the v2 version), and your auth tokens are in `gapi.auth2.getAuthInstance().currentUser.get().getAuthResponse()` This may have changed

**Comment by cjhowe7** [comment]
Having an unminified version helps a ton with debugging. I can minify it myself and include source maps this way, for example. When you have an error pointing to some minified code, it’s massively frustrating to not be able to debug without very careful Googling.

**Comment by Maxim-Mazurok** [comment]
There's a work in progress [Un-minified GAPI project](https://github.com/Maxim-Mazurok/gapi), feel free to check it out and contribute

**Comment by dvanderb** [comment]
This looks like a question about Google Login and Authentication. Please address this question to the google-signin tag on stackoverflow(https://stackoverflow.com/questions/tagged/google-signin)

If you believe this is in error, please reopen the issue.

**Comment by Maxim-Mazurok** [comment]
I believe this is an error, StackOverflow community can't release source code that Google owns. Also, I don't have permission to reopen the issue, @dvanderb, please reopen it for me :)

**Comment by callegar** [comment]
Indeed, this is not a question. It is more of a wishlist item or an enhancement request to how the code is distributed.

Minification serves a specific purpose, but is also a significant form of obfuscation of the source code. Shipping only the minified version of the code is certainly not the same as shipping the source code in a form suitable for understanding, debugging or modification. Even if it is probably OK under the Apache licence (that contrarily to other licenses does not explicitly demand that source is delivered in the form *preferred* for modification), it is certainly not very much in the spirit of providing source code that is practically open.

SiMilarLYWRitingenGlishInaformtHatthaTmAKEsitdiFFiculttOreADwoUln'TmaKevERyeFFeCTivecOmmuNicaTIon.
